README.md

```diff
@@ -644,6 +644,21 @@ AEMReporterIOS.logAEMEvent(eventName, value, currency, otherParameters);
 
 Event names for AEM must match event names you used in app event logging.
 
+Here's an example of how to use this method - 
+
+```ts
+LogFBPurchase = (purchaseAmount: number, currencyCode: string, parameters?: Params | undefined) => {
+    AppEventsLogger.logPurchase(purchaseAmount, currencyCode, parameters);
+    AEMReporterIOS.logAEMEvent("fb_mobile_purchase", purchaseAmount, currencyCode, parameters);
+}
+
+LogFBEvent = (eventName: string, ...args: Array<number | Params>) => {
+    AppEventsLogger.logEvent(eventName, ...args);
+    AEMReporterIOS.logAEMEvent(eventName, 0, undefined, {});
```

Comment:
The AEM event will lose parameters on this line, in our project we implement it as below

```ts
export function LogFBEvent(eventName, data){
  // we have a create function to make payload from data by event name
  const payload = makePayloadFromDataByEvent(eventName, data);
  const { value, parameters } = payload;

  // the original log event we have already implemented before AEM exists
  AppEventsLogger.logEvent(eventName, value, parameters);

  // additionally send AEM event with payload same with previous log event
  AEMReporter.logAEMEvent(eventName, value, parameters.fb_currency, parameters);
}
```

if we just look at the types, it's hard to sync the usage because the interface between `logEvent` and `logAEMEvent` is quite different,  but if we look into the implementation we can see they are quite the same

I encourage using the interface below for `logEvent`

```ts
logEvent(eventName: string, valueToSum: number, parameters: {[key:string]:string|number});
```

because the only difference between them is `currency`

```ts
logAEMEvent(eventName: string, value: number, currency?: string | undefined, otherParameters?: Record<string, string | number> | undefined)
```

so after all, the `LogFBEvent` can be changed to

```ts
LogFBEvent = (eventName: string, valueToSum: number, parameters: Record<string,string | number>)=> {
    AppEventsLogger.logEvent(eventName, valueToSum, parameters);
    AEMReporterIOS.logAEMEvent(eventName, valueToSum, parameters.fb_currency, parameters);
}
```

hope I'm clear enough, any feedback is welcome

Reply:
I like it! Changed!